File: Week 2/Readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: \n");

    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    float l = (letters * 1.0) / (words * 1.0) * 100;
    float s = (sentences * 1.0) / (words * 1.0) * 100;
    int index = round(0.0588 * l - 0.296 * s - 15.8);

    if (index >= 1 && index <= 16)
    {
        printf("Grade %i\n", index);
    }
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    if (index > 16)
    {
        printf("Grade 16+\n");
    }
}

int count_letters(string text)
{
    int i;
    int n = strlen(text);
    int counter = 0;
    for (i = 0; i < n; i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            counter++;
        }
        else
        {
            counter += 0;
        }
    }
    return counter;
}

int count_words(string text)
{
    int i;
    int n = strlen(text);
    int counter = 1;
    for (i = 0; i < n; i++)
    {
        if (text[i] == ' ')
        {
            counter++;
        }
        else
        {
            counter += 0;
        }
    }
    return counter;
}

int count_sentences(string text)
{
    int i;
    int n = strlen(text);
    int counter = 0;
    for (i = 0; i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            counter++;
        }
        else
        {
            counter += 0;
        }
    }
    return counter;
}

File: Week 6/Sentimental-readability/readability.py
# TODO

from cs50 import get_string

def main():
    text = get_string("Text: ")

    letters = float(count_letters(text))
    words = float(count_words(text))
    sentences = float(count_sentences(text))

    l = letters / words * 100
    s = sentences / words * 100
    index = int(round((0.0588 * l - 0.296 * s - 15.8),0))


    if (index >= 1 and index <= 16):
        print(f"Grade {index}")
    elif (index < 1):
        print("Before Grade 1")
    elif (index > 16):
        print("Grade 16+")

def count_letters(text):
    n = len(text)
    counter = 0
    for i in range(n):
        if ((text[i] >= 'a' and text[i] <= 'z') or (text[i] >= 'A' and text[i] <= 'Z')):
            counter += 1
        else:
            counter += 0
    return counter

def count_words(text):
    n = len(text)
    counter = 1
    for i in range(n):
        if (text[i] == " "):
            counter += 1
        else:
            counter += 0
    return counter


def count_sentences(text):
    n = len(text)
    counter = 0
    for i in range(n):
        if (text[i] == '.' or text[i] == '!' or text[i] == '?'):
            counter += 1
        else:
            counter += 0
    return counter

main ()
